infrastructure/code/code.go

```diff
@@ -64,6 +86,41 @@ func (sc *Scanner) SupportedCommands() []snyk.CommandName {
 }
 
 func (sc *Scanner) Scan(ctx context.Context, _ string, folderPath string) []snyk.Issue {
+	// When starting a scan for a folderPath that's already scanned, the new scan will wait for the previous scan
+	// to finish before starting.
+	// When there's already a scan waiting, the function returns immediately with empty results.
+	waitForPreviousScan := false
+	scanStatus := NewScanStatus()
+	scanStatus.isRunning = true
+	sc.scanStatusMutex.Lock()
```

Comment:
I may be missing something, but there should be a `defer sc.scanStatusMutex.Unlock()` so the function is never exited without releasing the lock.

Reply:
I'm not sure if we can use defer here because the mutex is unlocked on line 108, and there's a deferred function that locks and unlocks the mutes on lines 118-123.

I'll add an unlock before returning + tests